### PR TITLE
Changed controllers wildcards from static {one?}{two?}{three?}... to dynamic by ReflectionAPI

### PR DESCRIPTION
**Pull request by slavarazum**
Now we can correctly use route model bindings and pattern filters for controllers in **_RouteServiceProvider_**.
When we registered a controller:

``` php
Route::controller('some', 'SomeController');
```

with action:

``` php
public function getAction($foo, $bar, $baz = 'World!')
{
    return 'Hello ' . $baz;
}
```

Laravel will register _/some/action/{one?}/{two?}/{three?}/{four?}/{five?}_ route
Now it will look like **/some/action/{foo}/{bar}/{baz?}**
After we can use:

``` php
public function boot(Router $router)
{
    $router->pattern('foo', '[0-9]+');
    //or
    $router->model('bar', \App\User::class);
    //or
    $router->bind('baz', function ($value) {
            return strtoupper($value);
    });
    parent::boot($router);
}
```

### CODE REVIEW

File: src/Illuminate/Routing/ControllerInspector.php [hunk 1]
```diff
@@ -123,11 +123,16 @@ public function getPlainUri($name, $prefix)
     /**
      * Add wildcards to the given URI.
      *
-     * @param  string  $uri
+     * @param  string $uri
+     * @param ReflectionMethod $method
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/ControllerInspector.php]: cs

File: src/Illuminate/Routing/ControllerInspector.php [hunk 2]
```diff
@@ -123,11 +123,16 @@ public function getPlainUri($name, $prefix)
     /**
      * Add wildcards to the given URI.
      *
-     * @param  string  $uri
+     * @param  string $uri
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/ControllerInspector.php]: cs

**Comment by GrahamCampbell** [comment]
1. Please fix your cs and phpdoc.
2. This is a breaking change so must go to 5.2.

**Comment by Lamp27** [comment]
I will prepare it in a few minutes and send to the master branch.

**Pull request closed without merging**